### ISSUE TITLE
Cannot set multiple azurerm_security_center_contact ressource

### ISSUE THREAD

**Issue by sinc59** [hashicorp/terraform-provider-azurerm#18971]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.1

### AzureRM Provider Version

3.24.0

### Affected Resource(s)/Data Source(s)

azurerm_security_center_contact

### Terraform Configuration Files

```hcl
resource "azurerm_security_center_contact" "example" {
  email = "test@test.fr"
  phone = null

  alert_notifications = true
  alerts_to_admins    = true
}

resource "azurerm_security_center_contact" "example2" {
  email = "contact@example.com"
  phone = null

  alert_notifications = true
  alerts_to_admins    = true
}
```

### Debug Output/Panic Output

```shell
Terraform says:
Apply complete! Resources: 2 added, 0 changed, 0 destroyed

Azure cli says:
$ az security contact list
AlertNotifications    AlertsToAdmins    Email                           Location     Name      Phone
--------------------  ----------------  ------------------------------  -----------  --------  -------
On                    On                contact@example.com             West Europe  default1
```

### Expected Behaviour

Two contacts shloud be created.
The `azurerm_security_center_contact` does not have any name argument, he use an predefined name (default1).

I think that the `azurerm_security_center_contact` resource should have an name option (required, has the azure api specifications). 

### Actual Behaviour

Terraform seems to create the user with predefined name, when it's create the second user, the api override the first entry insted of create them.

### Steps to Reproduce

terraform apply

**Comment by neil-yechenwei** [comment]
Thanks for raising this issue. Per the [clarification](https://github.com/hashicorp/terraform-provider-azurerm/blob/e42208bec9663905bc0259494477a4b93edba90e/internal/services/securitycenter/security_center_contact_resource.go#L19), seems only `default1` is allowed by API design.

**Comment by sinc59** [comment]
Hello,
Yes, i don't know if the API have evolved but, has you see in the [Microsoft documentation](https://learn.microsoft.com/en-us/rest/api/defenderforcloud/security-contacts/create?tabs=HTTP) , the `securityContactName` variable is required and not hardcoded on default1.

Also, when i create a security contact with the azure cli, i can set a custom name (required also).

I think that we can try to add a name variable and avoid the hardcoded "default1", (if the azure cli works, the api rest should works as the same (:crossed_fingers: ))

**Comment by stephybun** [comment]
Closed by #18999
